### ISSUE TITLE
What about Cyrillic glyphs?

### ISSUE THREAD

**Issue by Twilightssuperb** [erikdkennedy/figtree#6]
Is it possible to create and implement cyrillic languages support?

**Comment by erikdkennedy** [comment]
I'm open to it. There are many improvements I need to make to the Latin set first (first and foremost, italics), so it won't be soon. But I'll leave this open for now, and others interested can upvote/comment.

**Comment by drawcard** [comment]
I'm interested in helping out with this. I've made a start at https://github.com/drawcard/figtree/tree/cyrillic

**Comment by drawcard** [comment]
I've completed the uppercase and lowercase character set for Cyrillic (Basic set), you can check out my branch to take a closer look.

I'm not a native Cyrillic writer, so it should be looked over by someone who is more familiar with Cyrillic. However I did study quite a few Cyrillic fonts while working on it, so hopefully it hits the mark.

Things to do before PR:
- Finish kerning pairs (will do some small updates to my branch as this happens)
- Check that the weight / colour is consistent with latin
- Final pass to check for technical issues with linework

Future ideas:
- Support for Ukranian, Belarusian, Bulgarian characters
- Italic versions

<img width="1230" alt="CleanShot 2022-10-04 at 23 19 41@2x" src="https://user-images.githubusercontent.com/1435638/193823546-d8c843df-a027-4210-a539-400da4379b74.png">


<img width="1228" alt="CleanShot 2022-10-04 at 23 14 30@2x" src="https://user-images.githubusercontent.com/1435638/193823487-f9163f6a-fc51-4da5-a759-cec0999970b8.png">

**Comment by erikdkennedy** [comment]
Looks like a fantastic start! Let me try and get some native readers to look it over. Thanks @drawcard ! 🙂

**Comment by erikdkennedy** [comment]
@drawcard I'd like to include Ukrainian with the initial PR.

Ґ ґ Є є І і Ї ї ₴

What else do we need for that?

**Comment by drawcard** [comment]
Thanks Erik!

For Ukrainian, I can complete the character set in Glyphs and update the branch. Shouldn't be too much effort to do.

I also posted in TypeDrawers to get some feedback, the comments have been helpful and I'll be working in those suggestions.

Will submit my PR soon!

**Comment by Twilightssuperb** [comment]
@drawcard, I add some line up to your mockup. It better shows that some letters don't have the same alignment (to bottom or top) and should be a bit smaller. Seems they are Э, Ю and small «э», «ю». But I'm not sure of my viewpoint. Just noticed some details that could be matter.

Your mockup is fine.

My orange line is 2 px.

![Figtree Cyrillic lineage mockup](https://user-images.githubusercontent.com/7080928/194062375-54b1eb8d-43aa-4ece-bf56-c2bc9c1d6631.png)

**Comment by drawcard** [comment]
Thanks for this @Twilightssuperb - I'm not entirely sure if I understand the issue, as the rounded letter shapes have overshoots on them, for optical correction, so their extremes wouldn't line up with the other x-heights:

<img width="1160" alt="CleanShot 2022-10-05 at 23 29 11@2x" src="https://user-images.githubusercontent.com/1435638/194066198-04de35bc-5660-4088-9055-9ddc7fbbf515.png">

<img width="1194" alt="CleanShot 2022-10-05 at 23 29 44@2x" src="https://user-images.githubusercontent.com/1435638/194066333-d0b851ec-1926-4836-abea-de8ee3b8fa08.png">

Does that make sense, or does the issue lie somewhere else?

**Comment by Twilightssuperb** [comment]
Yep, I suggested it. It's not a bug or mistake, it's a feature. Your drafts looks fine, don't bother. 🙂

I'm not a pro or designer to be nerdy or too strict, I'm just a kind of geek.

**Comment by drawcard** [comment]
Oh, I see now - thanks for pointing this out anyway!

**Comment by lazarljubenovic** [comment]
Serbian always gets left out 😭 ЉЊШЂЖЧЋ 🤟🏻 

Here's my feedback for the current state:

- The lowercase л looks weird, especially in the bold variant. Visually it doesn't feel like the left part is reaching the baseline. The curve should either be snappier (e.g. Monserrat), or the whole left stroke should have a slight slope from the start (e.g. Open Sans).
- Дs look enormous. The baseline should be at the upper (not lower) edge of the lower horizontal line. (More obvious in the bold version).
- We write б slightly different in Serbia (see [this comparison](https://en.wikipedia.org/wiki/Serbian_Cyrillic_alphabet?useskin=vector#/media/File:Cyrillic_alternates.svg)), so take this with a grain of salt, but I feel like the suggested б is too "wavy" at the top for this font.
- Ж seems slightly too condensed to the middle. I'd try extending the middle horizontal line and thus slightly reducing the slope of the diagonal strokes.
- З looks too thin compared to everything else, but I have no concrete suggestion on what to do.

**Comment by comerc** [comment]
@drawcard I can't install Cyrillic font on Mac. Please help me.

**Comment by comerc** [comment]
![photo_2024-03-23_11-31-54](https://github.com/erikdkennedy/figtree/assets/1025241/9642b5cd-b6ac-4259-a3d7-676d465b3c20)

keycaps with this font :)

**Comment by thejupitergroup** [comment]
@drawcard I tried to download the fonts from your branch but they don't have the Cyrillic characters. Am I looking in the wrong place?

**Comment by jilotta** [comment]
Hey! Native Russian speaker here.

@drawcard the lower bars of `Д`/`д` should be on the baseline, yours are below. The top part of the `б` is, in fact, too wavy.

`Л`/`л` looks weird; the "trunk" of it shouldn't be hanging like that. Here's how Bahnschrift handles it, for example. You can also see the `д` and `б`:

![Image](https://github.com/user-attachments/assets/0e40431e-15a5-4b42-91c4-d54d614cd2ec)

**Comment by ServusJon** [comment]
@erikdkennedy will this ever be merged?

**Comment by erikdkennedy** [comment]
My understanding is that Google Fonts will review it and merge, cf. https://github.com/google/fonts/issues/8723#issuecomment-2808683207
